### PR TITLE
Added note about loosely-coupled components (to the assessment)

### PR DESCRIPTION
**Pull request by andyblundell**
Felt like this needed calling-out since it's an area of focus

### CODE REVIEW

File: assessment.md [hunk 1]
```diff
@@ -143,6 +143,7 @@ You may wish to score each individual component or system separately for these a
 ### 8. Tech and architecture
 * The tech helps us deliver value.
 * We enjoy working with it and it supports fast, reliable and safe delivery.
+* We build self-contained functionality as independent components.
```
Comment (ivorc) [inline assessment.md:146]: This is tricky one. We want to get across the idea clearly without appearing to be dogmatic. Maybe
```suggestion
* Our system is made up of the right number of independent components/services which have the right responsibilities.
```

**Review by ivorc — COMMENTED** [review]
Proposed alternate wording, though I'm not 100% happy with mine either!

**Comment by arctangent** [comment]
Tech doesn't exist in a vacuum: the decision as to where system/component boundaries should lie must also take into account the organisational structure and skills mix - see also Conway's Law, and consider also the cost of communication. Often a monolith is more efficient for a small multi-skilled team to maintain, whereas highly componentised (microservice) architecture better suits a large total number of developers divided into separate teams each with niche skillsets.

**Comment by stefaniuk** [comment]
@arctangent this is a valid point. Whether a modularisation (monolith) or componentisation (a distributed design) is appropriate it's probably a separate discussion that requires more understanding of the problem we are solving. They are both enablers if used correctly. However, if we focus on defining the domain bundaries and are able to do it right we are not going to go wrong with either. While the product/service grows and we get more insight into the business, subdomains and resulting bounded contexts are more clear. Then is the time to recognise and apply the reverse Conway maneuver. @andyblundell and @ivorc I grepped the repo for any reference to team topologies and found nothing. This might belong to a different workstream but that topic deserves at least a bullet point in here, if not a proper paragraph - so people are aware.

**Comment by andyblundell** [comment]
Thanks @ivorc, @arctangent and @stefaniuk - good discussion.

This is a difficult concept to represent at a suitable level here - I suspect the discussion will be easier (but still not easy - as per note from @stefaniuk this is not easy to think about without a real example) when we flesh this section out in the engineering principles. From my point of view, there are massive potential advantages from componentisation, but I agree it's counter-productive if done in the wrong way - and I agree with @stefaniuk that getting those boundaries right is key. 

Maybe a better (or at least simpler) angle for the assessment is whether the team is happy it's having the right sorts of discussions about componentisation, and that decisions around componentisation (or not) are being reached in an informed way.

Any thoughts on that? 

@arctangent - in terms of current constraints: our view of good engineering needs to be grounded, but it doesn't need to be fully achievable under current constraints - it will inform discussions about those concerns / constraints.

@stefaniuk - yes, that's being considered, but not within this repo (yet). Let's discuss.

**Comment by arctangent** [comment]
A necessary prerequisite for componentisation is proper separation of concerns within a product. So we should emphasise the latter property as the most important thing for a product team to focus on. Whether or not a loosely-coupled part of a product should be extracted as a component for reuse across products is maybe something that an architect across the domain should focus on? I guess what this response is saying is: who is reading our guidance and what message is appropriate for their level within the architectural hierarchy?

**Comment by andyblundell** [comment]
@ivorc @arctangent @stefaniuk - proposed new wording:

`We have informed discussions about when/whether to split functionality into independent components, and the components we build have clear and clean boundaries of responsibility`

File: assessment.md [hunk 2]
```diff
@@ -143,6 +143,7 @@ You may wish to score each individual component or system separately for these a
 ### 8. Tech and architecture
 * The tech helps us deliver value.
 * We enjoy working with it and it supports fast, reliable and safe delivery.
+* We have informed discussions about when/whether to split functionality into independent components, and the components we build have clear and clean boundaries of responsibility.
```
Comment (ivorc) [inline assessment.md:146]: Now we have a bit more detail in the Architect for Flow pattern we might be able to make this a bit snappier, maybe:
```suggestion
* Our system is built as a set of independent services/components where appropriate (see [Architect for Flow](patterns/architect-for-flow.md)).
```

**Review by ivorc — APPROVED** [review]

**Pull request merged**